src/materials/MeshRefractionMaterial.tsx

```diff
@@ -130,10 +124,15 @@ export const MeshRefractionMaterial = shaderMaterial(
   #endif
   
   void main() {
-    mat4 modelMatrixInverse = inverse(modelMatrix * vInstanceMatrix);
+    #ifdef USE_INSTANCING
+      mat4 modelMatrixInverse = inverse(modelMatrix * vInstanceMatrix);
```

Comment:
I missed this before but this inverse in the fragment shader is also bad...modelMatrixInverse could be moved to the vertex shader and then you wouldn't have to pass vInstanceMatrix at all (except it would be always passing the varying instead of conditionally passing it...probably still worth it to avoid the inverse)